pulldown-cmark/src/parse.rs

```diff
@@ -617,14 +618,97 @@ impl<'input, F: BrokenLinkCallback<'input>> Parser<'input, F> {
                         ty: LinkStackTy::Image,
                     });
                 }
-                ItemBody::MaybeLinkClose(could_be_ref) => {
+                ItemBody::MaybeLinkClose(double, could_be_ref) => {
                     self.tree[cur_ix].item.body = ItemBody::Text {
                         backslash_escaped: false,
                     };
                     if let Some(tos) = self.link_stack.pop() {
                         if tos.ty == LinkStackTy::Disabled {
                             continue;
                         }
+                        if tos.ty == LinkStackTy::Link(true) {
```

Comment:
Ordinary markdown links are allowed to start or end with `[[` and `]]`, without the other side also being doubled up (meaning it's not a wikilink).

For example:

```markdown
[[squid] calamari is considered a delicacy](https://en.wikipedia.org/wiki/Squid)

[calamari [squid]](https://en.wikipedia.org/wiki/Squid)
```

> [[squid] calamari is considered a delicacy](https://en.wikipedia.org/wiki/Squid)
>
> [calamari [squid]](https://en.wikipedia.org/wiki/Squid)

This works in commonmark-hs with wikilinks turned on ([playground](https://pandoc.org/try/?params={%22text%22%3A%22[[squid]+calamari+is+considered+a+delicacy](https%3A%2F%2Fen.wikipedia.org%2Fwiki%2FSquid)\n\n[calamari+[squid]](https%3A%2F%2Fen.wikipedia.org%2Fwiki%2FSquid)%22%2C%22to%22%3A%22html5%22%2C%22from%22%3A%22commonmark%2Bwikilinks_title_after_pipe%22%2C%22standalone%22%3Afalse%2C%22embed-resources%22%3Afalse%2C%22table-of-contents%22%3Afalse%2C%22number-sections%22%3Afalse%2C%22citeproc%22%3Afalse%2C%22html-math-method%22%3A%22plain%22%2C%22wrap%22%3A%22auto%22%2C%22highlight-style%22%3Anull%2C%22files%22%3A{}%2C%22template%22%3Anull})), and should work in any implementation that parses wikilinks in a separate pass (assuming they correctly "bail out" if none are found).

I think that you should consider using a separate pass.